### PR TITLE
refactor: Convert SQFormInclusionListItem to TS Ref #244

### PR DESCRIPTION
**Pull request by sohail-panezai**
https://www.loom.com/share/005264408afe47ef9da042fcce148a1f

### CODE REVIEW

File: src/components/SQForm/SQFormInclusionListItem.tsx [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
+  /** Disabled state of the checkbox */
+  isDisabled?: boolean;
+  /** Descriptive label text for the checkbox */
+  label: string;
+  /** name of the checkbox array in `initialValues` */
+  name: string;
+  /** Custom onChange event callback */
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  /** Size of the input given full-width is 12. */
+  size?: 'auto' | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
```
Comment (caleboleary) [inline src/components/SQForm/SQFormInclusionListItem.tsx:20]: There's a `GridSize` type provided in MUI - you can see it used in `SQFormCheckboxGroup.tsx` - credit to @20BBrown14 for showing me this a couple of weeks ago.

Comment (codyarose) [reply src/components/SQForm/SQFormInclusionListItem.tsx:20]: You can also extend `BaseFieldProps`, then you won't need to specify the types for `name`, `label`, and `size`
```ts
interface SQFormInclusionListItemProps extends BaseFieldProps {
  /** evaluation of whether the box should be checked */
  isChecked: boolean;
  /** Disabled state of the checkbox */
  isDisabled?: boolean;
  /** Custom onChange event callback */
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
```

**Review by codyarose — CHANGES_REQUESTED** [review]

**Comment by codyarose** [comment]
Make sure you're using `npm run commit` to commit your changes. It helps with keeping the changelog and versioning accurate. Some more details about that [here](https://github.com/SelectQuoteLabs/SQForm#contributing).

File: src/components/SQForm/SQFormInclusionListItem.tsx (hunk 1 above)
Comment (sohail-panezai) [reply src/components/SQForm/SQFormInclusionListItem.tsx:20]: Great suggestions guys. Thanks for the feedback. Making those changes.

**Review by caleboleary — APPROVED** [review]

**Review by SeanGroff — APPROVED** [review]

**Pull request merged**